text2sql-backend/dataline/services/settings.py

```diff
@@ -62,10 +69,28 @@ async def update_user_info(self, session: AsyncSession, data: UserUpdateIn) -> U
         if user_info is None:
             # Create user with data
             user_create = UserCreate.model_construct(**data.model_dump(exclude_none=True))
+            if user_create.openai_api_key and user_create.preferred_openai_model is None:
+                user_create.preferred_openai_model = (
+                    config.default_model
+                    if model_exists(user_create.openai_api_key, config.default_model)
+                    else "gpt-3.5-turbo"
+                )
             user = await self.user_repo.create(session, user_create)
         else:
             # Update user with data
             user_update = UserUpdate.model_construct(**data.model_dump(exclude_none=True))
+            if user_update.openai_api_key:
+                key_to_check = user_update.openai_api_key
+                model_to_check = (
+                    user_update.preferred_openai_model or user_info.preferred_openai_model or config.default_model
+                )
+                assert model_exists(
```

Comment:
We should not use asserts in production code. 

IN the future we may build a binary in OPTIMIZED mode which removes all asserts. 

Just use an if statement instead!

Reply:
True, I thought about the assertion in prod thing but I didn't think of us turning the code into binary in the future. Changes applied 👍